### PR TITLE
New proposal add validation

### PR DESCRIPTION
**Pull request by BenjaminMeijs**
# Description

Added form validation / restrictions for the following things:
Step 3: A proposal may only have at most 1 MintToken action.
Step 3 > MintToken action: There may not be any duplicate addresses, as these can simply be joined.
Step 4: As time may have passed since step 2, if  it must again be validated that the proposal start time is at least 10 minutes (or in this case, 5 minutes) from now.
Step 1 > resources. Add regex to ensure that url's are valid.

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/components/newProposal/actions/MintTokensInput.tsx [hunk 1]
```diff
@@ -137,44 +146,86 @@ const MintListItem = ({
...
+      </div>
+    </>
+  );
+};
+
+/**
+ * Performs the same logical task as the standard some method for lists.
+ * However, it does not test the entire list, but from the start (index 0) untill (exclusive) the given index.
+ */
+function someUntill<T>(
```
Comment (mik-0) [inline src/components/newProposal/actions/MintTokensInput.tsx:220]: Typo 🤓

Comment (ThijnK) [reply src/components/newProposal/actions/MintTokensInput.tsx:220]: Also put this in the `utils.ts` file instead

File: src/components/newProposal/steps/Metadata.tsx [hunk 2]
```diff
@@ -176,18 +178,21 @@ export const Metadata = () => {
...
 const ResourceInput = ({
   resource,
   onChange,
   onRemove,
   register,
   prefix,
+  errors,
 }: {
-  resource: { name: string; url: string };
+  resource: Resource;
```
Comment (ThijnK) [inline src/components/newProposal/steps/Metadata.tsx:190]: Use `ProposalResource` instead (already being imported in this file)

File: src/components/newProposal/steps/Metadata.tsx
Comment (ThijnK) [inline src/components/newProposal/steps/Metadata.tsx:1]: Check for empty resource name as well

**Review by ThijnK — CHANGES_REQUESTED** [review]
LGTM apart from some small changes

**Review by ThijnK — APPROVED** [review]
LGTM

**Pull request merged**